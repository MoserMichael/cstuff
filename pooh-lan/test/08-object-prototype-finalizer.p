# test where object has a finalizer - it is called when object is freed.


a := { 'field' : 42, 
       'finalizer' : sub ()
                        println( ~msg 'finalizer is called' )
                     end
     }  



# assigning a different value to a; the previous value is now freed and finalizer is called.
a = {}

