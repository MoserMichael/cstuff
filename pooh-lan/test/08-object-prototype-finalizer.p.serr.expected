004|a := { 'field' : 42 , 'finalizer' : sub () }
005|sub ()
006| println( ~msg 'finalizer is called' )...
005|end
013|a = {  }
